src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java

```diff
@@ -84,4 +85,15 @@
         @Param("maxFailureCount") int maxFailureCount,
         Pageable pageable
     );
+
+    // `notification sent` should be the final state for `metadata failure` and `signature failure`.
+    // including all as notification itself can fail too
+    @Query(
+        nativeQuery = true,
+        value = "SELECT COUNT(1) AS incomplete\n"
+            + "FROM envelopes\n"
+            + "WHERE DATE(createdat) < :date\n"
+            + "  AND status NOT IN ('METADATA_FAILURE', 'SIGNATURE_FAILURE', 'NOTIFICATION_SENT', 'COMPLETED')"
```

Comment:
I think we should just compare to `COMPLETED` and everything that isn't in this state, should contribute to the count. `NOTIFICATION_SENT` should be taken into account, as it marks the point at which the envelope was sent to the orchestrator via queue (a lot can go wrong from here) and the failure statuses probably can't even exist for an envelope, because they represent situations in which an envelope wouldn't be created.

Reply:
Failures such as metafile validation or signature error should not be considered as we would have notified Exela for those failures. Only files which are not processed because it failed due to doc upload or message was sent to queue but not completed should be reported in my opinion.

Reply:
i don't think there's going back from those states. there has to be a new envelope

Reply:
> Failures such as metafile validation or signature error should not be considered as we would have notified Exela for those failures. Only files which are not processed because it failed due to doc upload or message was sent to queue but not completed should be reported in my opinion.

Can envelopes even exist with those statuses? I don't think we'll create them in the DB in case of validation/signature failure. We don't need to list them here. The only status that should remain is `COMPLETED` - everything that's in a different one for too long is definitely a problem.

Reply:
Good point @lgonczar yeah we create it only in events table. I forgot that we have events and envelope table. Yes query can be just !COMPLETED.

Reply:
lol. back to the beginning